platform/javascript/detect.py

```diff
@@ -1,8 +1,11 @@
 import os
+import sys
 
 from emscripten_helpers import run_closure_compiler, create_engine_file, add_js_libraries
 from SCons.Util import WhereIs
 
+import methods
```

Comment:
Since you only use `get_compiler_version`, I'd suggest to use `from methods import get_compiler_version`.